src/sql/workbench/contrib/executionPlan/browser/executionPlanFileViewCache.ts

```diff
@@ -0,0 +1,22 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ExecutionPlanFileView } from 'sql/workbench/contrib/executionPlan/browser/executionPlanFileView';
+
+export class ExecutionPlanFileViewCache {
+	private static instance: ExecutionPlanFileViewCache;
```

Comment:
We shouldn't be using singleton patterns like this. If you need a singleton then it should be a service.

But realistically this seems like something that should be a part of the existing code instead, typically I would expect the model to be the thing handling the caching.

Reply:
At first I was storing it int the model but since the model is the commons folder it cannot refer to anything from browser. Therefore I had to create a separate caching class.

Reply:
This deserves a separate offline discussion I think. And definitely shouldn't be done as part of all this other refactoring if you can avoid it.

Reply:
There is probably a better pattern for caching these results.  However, for April this is fine if we need more time to workout the better design (i.e. this doesn't appear to have risk of side-effects, etc.).  If you do merge as-is for April please create an issue in June to follow-up.